src/main/java/com/atherys/towns/service/NationService.java

```diff
@@ -106,6 +108,11 @@ public void addTown(Nation nation, Town town) {
         nationRepository.saveOne(nation);
     }
 
+    public Set<Player> getNationOnlinePlayers(Nation nation) {
+        return nation.getTowns().stream().flatMap(
+                town -> townService.getTownOnlinePlayers(town).stream()).collect(Collectors.toSet());
+    }
+
```

Comment:
There is already a method for this in the NationFacade, although that was written before the nation changes so its much more inefficient and should be replaced with this. Personally I also think this might fit better in the NationFacade since it isn't really doing any mutation.

Reply:
Moved to the NationFacade function, but replaced the implementation